auslib/dockerflow.py

```diff
@@ -29,6 +29,10 @@ def heartbeat():
         database_entry_value = heartbeat_database_fn(dbo)
         return Response(str(database_entry_value), headers={"Cache-Control": "no-cache"})
 
+    @app.errorhandler(502)
```

Comment:
Because we have control over the function that's returning the error, I don't think using app.errorhandler is the right thing to do. Instead, you can put a try/catch inside of the heartbeat function above, and return a 502 from in the exception case.

This wouldn't work anyways, because the current state of things means 500s are raised for errors, not 502s.

Once you do that, you should notice that the "testHeartbeatWithException" tests from https://github.com/mozilla/balrog/blob/master/auslib/test/admin/views/test_dockerflow.py and https://github.com/mozilla/balrog/blob/master/auslib/test/web/test_dockerflow.py#L25 start failing, which you'll need to update.

As for what to return in the body, the ideal thing is to return the exception name and message, but we need to be certain that there's no possibility of leaking SQLAlchemy credentials from doing so. Can you try to find out if SQLAlchemy ever shows database credentials in Exceptions before we go that route?